bitbots_vision/bitbots_vision/vision_modules/yoeo/vision_components.py

```diff
@@ -556,44 +557,32 @@ def set_image(self, image: np.ndarray) -> None:
         pass  # Nothing should happen here
 
 
-class YOEOObstacleDetectionComponent(IVisionComponent):
-    """
-    This component carries out the obstacle detection using YOEO.
+class RobotDetectionComponentTemplate(IVisionComponent):
     """
+        This template carries out the robot detection using YOEO. Abstract method "_configure_detectors" needs to be
+        implemented.
+        """
 
     def __init__(self, node: Node):
         self._config: Dict = {}
         self._debug_image: Optional[debug.DebugImage] = None
         self._debug_mode: bool = False
 
-        self._misc_obstacles_detector: Optional[obstacle.ColorObstacleDetector] = None
-        self._opponents_detector: Optional[obstacle.ColorObstacleDetector] = None
-        self._team_mates_detector: Optional[obstacle.ColorObstacleDetector] = None
+        self._misc_obstacles_detector: Optional[obstacle.CandidateFinder] = None
```

Comment:
In my opinion, it would make sense to add an abstract "set_image()"  to the CandidateFinder abstract class. As far as I have seen, most implementing classes provide such a method anyway and it would stop the type checker from complaining in RobotDetectionComponentTemplate.set_image(...)